### PR TITLE
Fix flaky test in ExecutionLifecycleTest

### LINKED ISSUE #2072: Flaky ExecutionLifecycleTest.when_psGetOnOtherNodeThrows_then_jobFails test
http://jenkins.hazelcast.com/job/jet-oss-pr-builder/2456/

Stacktrace:
```
org.junit.ComparisonFailure: 
expected exception not found in causes chain expected:<[java.lang.AssertionError: mock error]> but was:<[com.hazelcast.jet.JetException: java.lang.AssertionError: mock error
	at com.hazelcast.jet.impl.JobResult.getFailureAsThrowable(JobResult.java:105)
	at com.hazelcast.jet.impl.JobResult.asCompletableFuture(JobResult.java:117)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$callWithJob$34(JobCoordinationService.java:613)
	at com.hazelcast.jet.impl.JobCoordinationService.lambda$submitToCoordinatorThread$43(JobCoordinationService.java:997)
	at com.hazelcast.internal.util.executor.CompletableFutureTask.run(CompletableFutureTask.java:64)
	at com.hazelcast.internal.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:217)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80)
]>
	at org.junit.Assert.assertEquals(Assert.java:115)
	at com.hazelcast.jet.core.TestUtil.assertExceptionInCauses(TestUtil.java:79)
	at com.hazelcast.jet.core.ExecutionLifecycleTest.when_psGetOnOtherNodeThrows_then_jobFails(ExecutionLifecycleTest.java:242)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:114)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output:
```
Finished Running Test: when_executionCancelled_then_jobCompletedWithCancellationException in 0.026 seconds.
Started Running Test: when_psGetOnOtherNodeThrows_then_jobFails
2020-03-19 08:08:15,558 [ INFO] [hz.friendly_bose.cached.thread-17] [c.h.j.i.JobCoordinationService]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Starting job 0410-b44d-caeb-0001 based on submit request
2020-03-19 08:08:15,561 [ INFO] [hz.friendly_bose.cached.thread-17] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Didn't find any snapshot to restore for job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001
2020-03-19 08:08:15,562 [ INFO] [hz.friendly_bose.cached.thread-17] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Start executing job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001, execution graph in DOT format:
digraph DAG {
	"faulty" [localParallelism=72];
}
HINT: You can use graphviz or http://viz-js.com to visualize the printed graph.
2020-03-19 08:08:15,562 [DEBUG] [hz.friendly_bose.cached.thread-17] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Building execution plan for job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001
2020-03-19 08:08:15,563 [DEBUG] [hz.friendly_bose.cached.thread-17] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Built execution plans for job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001
2020-03-19 08:08:15,563 [DEBUG] [hz.friendly_bose.cached.thread-17] [c.h.j.i.o.InitExecutionOperation]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Initializing execution plan for job 0410-b44d-caeb-0001, execution 0410-b44d-caec-0001 from [127.0.0.1]:5701
2020-03-19 08:08:15,563 [DEBUG] [hz.jolly_bose.generic-operation.thread-16] [c.h.j.i.o.InitExecutionOperation]: [127.0.0.1]:5702 [jet] [4.1-SNAPSHOT] Initializing execution plan for job 0410-b44d-caeb-0001, execution 0410-b44d-caec-0001 from [127.0.0.1]:5701
2020-03-19 08:08:15,564 [ERROR] [hz.jolly_bose.generic-operation.thread-16] [c.h.j.i.o.InitExecutionOperation]: [127.0.0.1]:5702 [jet] [4.1-SNAPSHOT] mock error
java.lang.AssertionError: mock error
	at com.hazelcast.jet.core.ExecutionLifecycleTest.<clinit>(ExecutionLifecycleTest.java:84) ~[test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_241]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_241]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_241]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_241]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.12.jar:4.12]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) ~[junit-4.12.jar:4.12]
	at com.hazelcast.test.AbstractHazelcastClassRunner$1.evaluate(AbstractHazelcastClassRunner.java:301) ~[hazelcast-4.0.1-SNAPSHOT-tests.jar:4.0.1-SNAPSHOT]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292) ~[junit-4.12.jar:4.12]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_241]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_241]
2020-03-19 08:08:15,565 [ INFO] [hz.friendly_bose.cached.thread-17] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Execution plan for jobId=0410-b44d-caeb-0001, jobName='0410-b44d-caeb-0001', executionId=0410-b44d-caec-0001 initialized
2020-03-19 08:08:15,565 [DEBUG] [hz.friendly_bose.cached.thread-5] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Init of job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001 has failures: [[127.0.0.1]:5702=java.lang.AssertionError: mock error]
2020-03-19 08:08:15,565 [DEBUG] [hz.friendly_bose.cached.thread-5] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Sending CompleteExecutionOperation for job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001
2020-03-19 08:08:15,565 [DEBUG] [hz.friendly_bose.cached.thread-5] [c.h.j.i.o.CompleteExecutionOperation]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Completing execution 0410-b44d-caec-0001 from caller [127.0.0.1]:5701, error=java.lang.AssertionError: mock error
2020-03-19 08:08:15,565 [DEBUG] [hz.friendly_bose.cached.thread-5] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Completed execution of job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001
2020-03-19 08:08:15,565 [DEBUG] [hz.jolly_bose.generic-operation.thread-17] [c.h.j.i.o.CompleteExecutionOperation]: [127.0.0.1]:5702 [jet] [4.1-SNAPSHOT] Completing execution 0410-b44d-caec-0001 from caller [127.0.0.1]:5701, error=java.lang.AssertionError: mock error
2020-03-19 08:08:15,565 [DEBUG] [hz.jolly_bose.generic-operation.thread-17] [c.h.j.i.JobExecutionService]: [127.0.0.1]:5702 [jet] [4.1-SNAPSHOT] Completed execution of job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001
2020-03-19 08:08:15,566 [ERROR] [hz.friendly_bose.cached.thread-35] [c.h.j.i.MasterJobContext]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] Execution of job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001 failed
	Start time: 2020-03-19T08:08:15.558
	Duration: 8 ms
	For further details enable JobConfig.storeMetricsAfterJobCompletion
java.lang.AssertionError: mock error
	at com.hazelcast.jet.core.ExecutionLifecycleTest.<clinit>(ExecutionLifecycleTest.java:84) ~[test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_241]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_241]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_241]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_241]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.12.jar:4.12]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27) ~[junit-4.12.jar:4.12]
	at com.hazelcast.test.AbstractHazelcastClassRunner$1.evaluate(AbstractHazelcastClassRunner.java:301) ~[hazelcast-4.0.1-SNAPSHOT-tests.jar:4.0.1-SNAPSHOT]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:298) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:292) ~[junit-4.12.jar:4.12]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_241]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_241]
2020-03-19 08:08:15,567 [DEBUG] [hz.friendly_bose.cached.thread-35] [c.h.j.i.JobCoordinationService]: [127.0.0.1]:5701 [jet] [4.1-SNAPSHOT] job '0410-b44d-caeb-0001', execution 0410-b44d-caec-0001 is completed
BuildInfo right after when_psGetOnOtherNodeThrows_then_jobFails(com.hazelcast.jet.core.ExecutionLifecycleTest): BuildInfo{version='4.0.1-SNAPSHOT', build='20200317', buildNumber=20200317, revision=7ce74e7, enterprise=false, serializationVersion=1, jet=JetBuildInfo{version='4.1-SNAPSHOT', build='20200319', revision='a3e484b'}}
Hiccups measured while running test 'when_psGetOnOtherNodeThrows_then_jobFails(com.hazelcast.jet.core.ExecutionLifecycleTest):'
08:08:15, accumulated pauses: 185 ms, max pause: 179 ms, pauses over 1000 ms: 0
```

### PR DESCRIPTION
**Pull request by frant-hartm**
Fix flaky test where job can be completed from JobResult

Reproducible by adding delay to doInvokeJoinJob in AbstractJobProxy.
Fixes #2072

Add stacktrace to failureText in JobResult, stack trace is useful for debugging

Checklist
- [x] Tags Set
- [x] Milestone Set

### CODE REVIEW

File: hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/TestUtil.java [hunk 1]
```diff
@@ -69,7 +69,9 @@ public static void assertExceptionInCauses(final Throwable expected, final Throw
         boolean found = false;
         Throwable t = caught;
         while (!found && t != null) {
-            found = Objects.equals(t.getMessage(), expected.getMessage()) && t.getClass() == expected.getClass();
+            found = Objects.equals(t.getMessage(), expected.getMessage()) && t.getClass() == expected.getClass() ||
+                    (t.getMessage().contains(expected.getMessage()) &&
+                            t.getMessage().contains(expected.getClass().getName()));
```
Comment (viliam-durina) [inline hazelcast-jet-core/src/test/java/com/hazelcast/jet/core/TestUtil.java:74]: This deserves a specification in the javadoc.

**Comment by cangencer** [comment]
verify

**Comment by cangencer** [comment]
verify

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed ([log](
https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/2688/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/jet-oss-pr-builder/2688/jet-oss-pr-builder-2688.zip)). 
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   JobSummaryTest.when_job_failed:178 expected:<...t.jet.JetException: [2 of 5 tasklets failed to initialize. One of the failures is attached as the cause and its summary is java.lang.UnsupportedOperationException: Cannot subscribe to event journal because it is either not configured or disabled for map 'invalid']> but was:<...t.jet.JetException: [ 2 of 5 tasklets failed to initialize. One of the failures is attached as the cause and its summary is java.lang.UnsupportedOperationException: Cannot subscribe to event journal because it is either not configured or disabled for map 'invalid'
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.awaitAll(TaskletExecutionService.java:221)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.submitCooperativeTasklets(TaskletExecutionService.java:189)
	at com.hazelcast.jet.impl.execution.TaskletExecutionService.beginExecute(TaskletExecutionService.java:148)
	at com.hazelcast.jet.impl.execution.ExecutionContext.beginExecution(ExecutionContext.java:167)
	at com.hazelcast.jet.impl.JobExecutionService.beginExecution(JobExecutionService.java:377)
	at com.hazelcast.jet.impl.operation.StartExecutionOperation.doRun(StartExecutionOperation.java:46)
	at com.hazelcast.jet.impl.operation.AsyncOperation.run(AsyncOperation.java:53)
	at com.hazelcast.spi.impl.operationservice.Operation.call(Operation.java:184)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.call(OperationRunnerImpl.java:228)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:217)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:433)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:166)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:136)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.run(OperationThread.java:123)
Caused by: java.lang.UnsupportedOperationException: Cannot subscribe to event journal because it is either not configured or disabled for map 'invalid'
	at com.hazelcast.map.impl.journal.MapEventJournalSubscribeOperation.innerBeforeRun(MapEventJournalSubscribeOperation.java:52)
	at com.hazelcast.map.impl.operation.MapOperation.beforeRun(MapOperation.java:101)
	at com.hazelcast.spi.impl.operationservice.impl.OperationRunnerImpl.run(OperationRunnerImpl.java:215)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:160)
	at com.hazelcast.spi.impl.operationexecutor.impl.OperationThread.process(OperationThread.java:138)
	... 1 more
]>
[INFO] 
[ERROR] Tests run: 2147, Failures: 1, Errors: 0, Skipped: 11
[INFO] 

[ERROR] There are test failures.
</details>
</pre>

**Pull request merged**